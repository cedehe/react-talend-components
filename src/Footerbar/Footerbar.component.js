import React, { PropTypes } from 'react';
import { Nav } from 'react-bootstrap';

import Action from '../Actions/Action';
import theme from './Footerbar.scss';

function Footerbar({ main, secondary, cancel }) {
	return (
		<Nav className={theme['tc-footerbar-container']}>
			{cancel ? (
				<Nav className="pull-left">
					<Action {...cancel} />
				</Nav>
			) : null}
			<Nav className="pull-right">
				{secondary ? (
					<Action {...Object.assign({ bsStyle: 'secondary' }, secondary)} />
				) : null}
				<Action className={theme['tc-footerbar-btn']} {...Object.assign({ bsStyle: 'info' }, main)} />
			</Nav>
		</Nav>
	);
}

Footerbar.propTypes = {
	main: PropTypes.shape(Action.propTypes).isRequired,
	secondary: PropTypes.shape(Action.propTypes),
	cancel: PropTypes.shape(Action.propTypes),
};

export default Footerbar;
