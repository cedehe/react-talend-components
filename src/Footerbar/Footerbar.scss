.tc-footerbar-container {
	padding: $padding-small $padding-normal;
	background-color: $concrete;
	border-top: 1px solid rgba(0, 0, 0, .25 ) ;
}

.tc-footerbar-btn {
	margin-left: $padding-large;
}
