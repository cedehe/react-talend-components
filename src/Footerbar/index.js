import Footerbar from './Footerbar.component';

export default Footerbar;
