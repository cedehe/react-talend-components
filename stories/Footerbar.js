import React from 'react';
import { storiesOf, action } from '@kadira/storybook';

import { Footerbar } from '../src/index';

const footerbar = {
	main: {
		label: 'save',
		onClick: action('click main button'),
	},
	secondary: {
		label: 'save and add another',
		onClick: action('click secondary button'),
	},
	cancel: {
		label: 'cancel',
		onClick: action('click third button'),
	},
};

storiesOf('Footerbar', module)
	.addWithInfo('default', () => (
		<div>
			<h1>Footerbar</h1>
			<h2>Definition</h2>
			<p>Footer bar is similar to the action toolbar,
				but is placed below forms, panels or dialogs.
				<a href="http://guidelines.talend.com/document/92132#/navigation-layout/toolbars/action-bar">Spec</a>
			</p>
			<h2>Examples</h2>
			<div style={{ width: '600px' }}>
				<Footerbar {...footerbar} />
			</div>
		</div>
	));
