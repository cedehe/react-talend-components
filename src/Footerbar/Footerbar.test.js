import React from 'react';
import renderer from 'react-test-renderer';

import Footerbar from './Footerbar.component';

jest.mock('react-dom');

const props = {
	main: {
		label: 'save',
		onClick: jest.fn(),
	},
	secondary: {
		label: 'save and add another',
		onClick: jest.fn(),
	},
	cancel: {
		label: 'close',
		onClick: jest.fn(),
	}
};

describe('Footerbar', () => {
	it('should render 3 actions', () => {

		const wrapper = renderer.create(<Footerbar {...props} />);

		expect(wrapper).toMatchSnapshot();
	})
});
