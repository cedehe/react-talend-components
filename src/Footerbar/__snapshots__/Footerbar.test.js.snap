exports[`Footerbar should render 3 actions 1`] = `
<ul
  className="nav"
  role={null}>
  <ul
    active={undefined}
    className="pull-left nav"
    role={null}>
    <button
      className="btn btn-default"
      disabled={false}
      onClick={[Function]}
      role={null}
      type="button">
      <span>
        close
      </span>
    </button>
  </ul>
  <ul
    active={undefined}
    className="pull-right nav"
    role={null}>
    <button
      className="btn btn-default"
      disabled={false}
      onClick={[Function]}
      role={null}
      type="button">
      <span>
        save and add another
      </span>
    </button>
    <button
      className="btn btn-info"
      disabled={false}
      onClick={[Function]}
      role={null}
      type="button">
      <span>
        save
      </span>
    </button>
  </ul>
</ul>
`;
